src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs

```diff
@@ -264,5 +308,16 @@ private bool IsGoCliManuallyEnabled()
         {
             return EnvVarService.DoesEnvironmentVariableExist("EnableGoCliScan");
         }
+
+        private class GoBuildModule
+        {
+            public string Path { get; set; }
+
+            public bool Main { get; set; }
```

Comment:
Are the non-nullable types here guaranteed to exist?

Reply:
Yes to most of them, `Main` property could be missing, because Go doesn't include in their serialization the default value (false).